astronomer/airflow/version_check/update_checks.py

```diff
@@ -457,11 +465,35 @@ def available_eol(self) -> dict[str, Any] | None:
         )
         return self.get_eol_notice(current_version)
 
+    def available_yanked(self) -> dict[str, Any] | None:
+        """Check if the current version of Astronomer Runtime is yanked."""
+        from .models import AstronomerAvailableVersion
+
+        session = get_state(app=current_app).db.session
+        runtime_version = version.parse(get_runtime_version())
+        current_version = (
+            session.query(AstronomerAvailableVersion)
+            .filter(AstronomerAvailableVersion.version == str(runtime_version))
+            .one_or_none()
+        )
+
+        if current_version and current_version.yanked:
+            return {
+                "version": current_version.version,
+                "app_name": "Astronomer Runtime",
+                "description": "Warning: The current version {} of Astronomer Runtime has been yanked.".format(
```

Comment:
cc @cmarteepants on wording.

Probably better with something like this though:

```suggestion
                "description": "Warning: This version of Astronomer Runtime, {}, has been yanked.".format(
```

Reply:
We might want to link to the docs on yanked too. Maybe https://www.astronomer.io/docs/astro/runtime-version-lifecycle-policy#restricted-runtime-versions

Reply:
Added the link to docs as well